packages/valory/skills/register_termination_abci/composition.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the register-reset ABCI application."""
+
+import packages.valory.skills.registration_abci.rounds as RegistrationAbci
+import packages.valory.skills.reset_pause_abci.rounds as ResetAndPauseAbci
+from packages.valory.skills.abstract_round_abci.abci_app_chain import (
+    AbciAppTransitionMapping,
+    chain,
+)
+from packages.valory.skills.termination_abci.rounds import (
+    BackgroundRound,
+    Event,
+    TerminationAbciApp,
+)
+
+
+abci_app_transition_mapping: AbciAppTransitionMapping = {
+    RegistrationAbci.FinishedRegistrationRound: ResetAndPauseAbci.ResetAndPauseRound,
+    RegistrationAbci.FinishedRegistrationFFWRound: ResetAndPauseAbci.ResetAndPauseRound,
+    ResetAndPauseAbci.FinishedResetAndPauseRound: ResetAndPauseAbci.ResetAndPauseRound,
+    ResetAndPauseAbci.FinishedResetAndPauseErrorRound: ResetAndPauseAbci.ResetAndPauseRound,
+}
+
+RegisterTerminateAbciApp = chain(
+    (
+        RegistrationAbci.AgentRegistrationAbciApp,
+        ResetAndPauseAbci.ResetPauseABCIApp,
+    ),
+    abci_app_transition_mapping,
+).add_termination(
+    background_round_cls=BackgroundRound,
+    termination_event=Event.TERMINATE,
+    termination_abci_app=TerminationAbciApp,
+)
```

Comment:
We can define a `with_termination(user_abci_app)` util function that sets the termination related attrs to a user defined abci app. I think this would be helpful for the user because the `.add_termination()` args are always the same.